Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1202,48 +1202,75 @@ namespace Babylon
             XRRay(const Napi::CallbackInfo& info)
                 : Napi::ObjectWrap<XRRay>{info}
             {
-                bool originSet = false;
-                bool directionSet = false;
-                bool matrixSet = false;
-                if (info[0].IsObject())
+                auto argLength = info.Length();
+                xr::Ray tempVals{};
+                tempVals.Direction.Z = -1.0;
+
+                m_origin = Napi::Persistent(Napi::Object::New(info.Env()));
+                m_direction = Napi::Persistent(Napi::Object::New(info.Env()));
+                m_matrix = Napi::Persistent(Napi::Float32Array::New(info.Env(), MATRIX_SIZE));
```

Comment:
`m_matrix` isn't being set anymore. Is that expected?

Reply:
We didn't have any code in BJS that passed in a matrix on creation, and it doesn't look like there's any code that tries to access the matrix either. The docs have it initialized on query, so I think since nothing queries it it should be fine.

Reply:
It's okay to not implement, but it might be good to assert or throw an exception when it is passed in, just in case the Babylon.js side changes. Also, if it really isn't needed, then `m_matrix` isn't necessary.